src/silo/append/database_inserter.cpp

```diff
@@ -0,0 +1,53 @@
+#include "silo/append/database_inserter.h"
+#include "silo/schema/duplicate_primary_key_exception.h"
+
+namespace silo::append {
+
+void TablePartitionInserter::insert(const nlohmann::json& ndjson_line) {
+   for (auto& column_metadata : table_partition->columns.metadata) {
+      try {
+         table_partition->columns.addJsonValueToColumn(column_metadata, ndjson_line);
+      } catch (const nlohmann::json::type_error& error) {
+         throw silo::append::AppendException(
+            "The following line does not conform to SILO's json specification when adding "
+            "to database column {}: '{}'\n"
+            "We got a json type_error: {}",
+            column_metadata.name,
+            ndjson_line.dump(),
+            error.what()
+         );
```

Comment:
I don't see a red unit test when I comment out this catch block -> Let's add a test (or maybe even several ones?)